tests/app/main/test_briefs.py

```diff
@@ -1900,110 +1958,289 @@ def test_view_response_result_page_escapes_essential_and_nice_to_have_requiremen
 
         assert res.status_code == 200
         doc = html.fromstring(res.get_data(as_text=True))
-        assert doc.xpath('//h1')[0].text.strip() == "You don’t meet all the essential requirements"
 
-    def test_view_response_result_not_submitted_redirect_to_submit_page(self, data_api_client):
+        requirements_data = self._get_data_from_table(doc, "Your details")
+        assert requirements_data.exists()
+        assert requirements_data.row(0).cell(0) == "Day rate"
+        assert requirements_data.row(0).cell(1) == "£300"
+        assert requirements_data.row(1).cell(0) == "Date the specialist can start work"
+        assert requirements_data.row(1).cell(1) == "02/02/2017"
+        assert requirements_data.row(2).cell(0) == "Email address"
+        assert requirements_data.row(2).cell(1) == "contact@big.com"
+
+    def test_nice_to_haves_heading_not_shown_when_there_are_none(self, data_api_client):
+        self.set_framework_and_eligibility_for_api_client(data_api_client)
+        data_api_client.get_brief.return_value = self.brief
+        data_api_client.find_brief_responses.return_value = self.brief_responses
+        res = self.client.get('/suppliers/opportunities/1234/responses/result')
+
+        assert res.status_code == 200
+        doc = html.fromstring(res.get_data(as_text=True))
+        assert len(
+            doc.xpath('//h2[contains(normalize-space(text()), "Your nice-to-have skills and experience")]')
+        ) == 0
+
+    def test_budget_message_shown_if_budget_is_set_for_specialists(self, data_api_client):
+        brief_with_budget_range = self.brief.copy()
+        brief_with_budget_range['briefs']['budgetRange'] = 'Up to £200 per day'
+
+        self.set_framework_and_eligibility_for_api_client(data_api_client)
+        data_api_client.get_brief.return_value = brief_with_budget_range
+        data_api_client.find_brief_responses.return_value = self.brief_responses
+        res = self.client.get('/suppliers/opportunities/1234/responses/result')
+        assert res.status_code == 200
+        doc = html.fromstring(res.get_data(as_text=True))
+        assert len(doc.xpath('//li[contains(normalize-space(text()), "your day rate exceeds their budget")]')) == 1
+
+    def test_budget_message_not_shown_if_budget_is_not_set_for_specialists(self, data_api_client):
+        self.set_framework_and_eligibility_for_api_client(data_api_client)
+        data_api_client.get_brief.return_value = self.brief
+        data_api_client.find_brief_responses.return_value = self.brief_responses
+        res = self.client.get('/suppliers/opportunities/1234/responses/result')
+
+        assert res.status_code == 200
+        doc = html.fromstring(res.get_data(as_text=True))
+        assert len(doc.xpath('//li[contains(normalize-space(text()), "your day rate exceeds their budget")]')) == 0
+
+    def test_evaluation_methods_load_default_value(self, data_api_client):
```

Comment:
These two evaluation method tests appear to be exactly the same for both flows and I think can be moved into the shared tests.